src/libraries/Microsoft.PowerFx.Connectors/ConnectorFunction.cs

```diff
@@ -1174,8 +1176,25 @@ private async Task<ConnectorEnhancedSuggestions> GetConnectorSuggestionsFromDyna
                 foreach (JsonElement jElement in je.EnumerateArray())
                 {
                     JsonElement title = ExtractFromJson(jElement, cdv.ValueTitle);
-                    JsonElement value = ExtractFromJson(jElement, cdv.ValuePath);
-
+                    JsonElement value = ExtractFromJson(jElement, cdv.ValuePath);
+
+                    // Note: Some connectors have misalignment between the Swagger definition and the actual response, caused suggestion API
+                    // returning empty result. For example, Teams declares the GetMessageLocations response as List<{id, displayName}> but
+                    // in fact returns List<{value, displayName}>.
+                    // Fallback to "displayName" and "value" which are the most commonly used property names in suggestion response.
+                    if (connectorSettings != null && connectorSettings.AllowSuggestionMappingFallback)

```

Comment:
>connectorSettings != null && connectorSettings.AllowSuggestionMappingFallback [](http://example.com/codeflow?start=24&length=77)

nit
could be written `connectorSetting?s.AllowSuggestionMappingFallback == true`

Reply:
updated as we discussed